lib/res_controller.js

```diff
@@ -0,0 +1,47 @@
+class ResController {
```

Comment:
This class feels weird to me; it seems like some of its responsibilities (abstracting away the differences between a websocket and HTTP connection, such as communicating an error message to the user) are actually scattered in index.js. It's fine for now though.